src/main/ApplicationImpl.cpp

```diff
@@ -587,7 +587,7 @@ ApplicationImpl::manualClose(std::map<std::string, std::string> const& retMap,
         retStr = "ENHANCED_MANUAL_CLOSE not enabled";
         return;
     }
-    throw std::runtime_error("Enhanced manual close not implemented yet");
+    mHerder->triggerEnhancedManualCloseLedger(retMap, retStr);
```

Comment:
This doesn't seem to do all the same things as the other `manualClose`. Ideally these should share the same code path (after all, if `retMap.empty()` then they should be _identical_). You can either remove one, or have one invoke the other. 

Furthermore, as with `Herder::triggerEnhancedManualCloseLedger` this interface is not appropriate here. Handling the key-value pairs should be happening higher up in the call-stack.

Reply:
I've moved the key-value-pair handling to `CommandHandler.cpp`, and used the resulting restructuring to unify the `manualClose()` and `triggerNextLedger()` variants into single interfaces.